src/Octoshift/GithubApi.cs

```diff
@@ -304,5 +305,34 @@ public virtual async Task<bool> RevokeMigratorRole(string org, string actor, str
                 return false;
             }
         }
+
+        public virtual async Task<int> StartArchiveGeneration(string apiUrl, string org, object options)
+        {
+            var url = $"{apiUrl}/orgs/{org}/migrations";
+
+            var response = await _client.PostAsync(url, options);
+            var data = JObject.Parse(response);
+            return (int)data["id"];
+        }
+
+        public virtual async Task<GithubEnums.ArchiveMigrationStatus> GetArchiveMigrationStatus(string apiUrl, string org, int migrationId)
+        {
+            var url = $"{apiUrl}/orgs/{org}/migrations/{migrationId}";
+
+            var response = await _client.GetAsync(url);
+            var data = JObject.Parse(response);
+
+            var state = (string)data["state"];
+
+            return GithubEnums.StringToArchiveMigrationStatus(state);
+        }
+
+        public virtual async Task<string> GetArchiveMigrationUrl(string apiUrl, string org, int migrationId)
+        {
+            var url = $"{apiUrl}/orgs/{org}/migrations/{migrationId}/archive";
+
+            var response = await _client.GetNonSuccessAsync(url, HttpStatusCode.Found);
```

Comment:
I think this wouldn't be ideal because we actually want the response since the status code is what we're expecting. There's been some refactoring the client side, but I see this as a stylistic difference

Reply:
It is not a stylistic difference that I am suggesting, with what I suggested you would both get the response content and the status code with `SendAsync`. The reason I am against using `GetNonSuccessAsync` is as I said before, it is there to accommodate a bad api design on the GitHub side and it's actually a business rule that we want to apply here. All I am saying it to simply use `SendAsync` if we also want to get the status code back.
To further explain why this is a bad design, imagine a scenario where the same thing happens with POST or PUT, what would you do in those cases? Would you just go ahead and add another method to the client class?
Besides what i am suggesting is a common practice that you can for example see in another popular .NET http  client called [Restsharp](https://restsharp.dev/)